test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -47,6 +58,17 @@ public void NonGeneric_GetRequiredService_Throws_WhenNoServiceRegistered()
         }
 
         [Fact]
+        public void IRequiredServiceAware_NonGeneric_GetRequiredService_Throws_WhenNoServiceRegistered()
+        {
+            // Arrange
+            var serviceProvider = new RequiredServiceAwareServiceProvider();
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredService(typeof(IFoo)),
```

Comment:
Yup. Making it an explicit implementation would avoid accidentally referencing the method rather than the extension.